### ISSUE TITLE
incorrect substring behaviour when no length provided

### ISSUE THREAD

**Issue by 7opf** [HL7/fhirpath.js#43]
The substring function currently does not handle empty length params correctly

```
'abcdefg'.substring(3) // expect: 'defg', actual: 'abc'
```

Looks like the issue lies here:
https://github.com/HL7/fhirpath.js/blob/c5969040945ec1306fc7ad47752c4342e07c59e9/src/strings.js#L20

When length is `undefined`, `start + length` will be `NaN` which seems to be treated the same as -1 by the substring function.

So currently ommitting the length param behaves the same as passing -1 to the length param.

Workaround:

```
'abcdefg'.substring(3, length()) // expect: 'defg', actual: 'defg'
```

**Comment by plynchnlm** [comment]
I have confirmed the problem.  Thanks for the report.

**Comment by plynchnlm** [comment]
Fixed in 2.2.1.
